airbyte-workers/src/main/java/io/airbyte/workers/temporal/CancellationHandler.java

```diff
@@ -48,7 +48,7 @@ public void checkAndHandleCancellation(final Runnable onCancellationCallback) {
          * {@link TemporalUtils#withBackgroundHeartbeat} for where we actually send heartbeats to ensure
          * that we don't time out the activity.
          */
-        context.heartbeat(null);
+        activityContext.heartbeat(null);
       } catch (final ActivityCompletionException e) {
```

Comment:
@jrhizor Since the cancellation handler is being called from scheduled background threads to initiate heartbeats I think that it should also handle generic exceptions in case there are some transient errors such as networking errors. The current implementation fails to do that which will lead to the scheduled executor service swallowing the exception and stopping all further work of the thread, including scheduling it again.